### ISSUE TITLE
./configure script needs a refresh

### ISSUE THREAD

**Issue by orlitzky** [NagiosEnterprises/ndoutils#69]
I have a RISC-V system now but the `./configure` script in v2.1.4 is too old to recognize it:

```
 ./configure 
checking for a BSD-compatible install... /usr/bin/install -c
checking for perl... /usr/bin/perl
checking build system type... ./config.guess: unable to guess system type

This script, last modified 2006-01-02, has failed to recognize
the operating system you are using....
configure: error: cannot guess build type; you must specify one
```

I think in the future it would make more sense to generate the `./configure` script as part of the release process? There are a bunch of updates to `configure.ac` that `autoupdate` wants to make for autoconf-2.72, too.

**Comment by aaronagios** [comment]
@orlitzky I'm a manager at Nagios, and I can read enough c to have approved a batch of your PRs (thank you for those) but I'm not the developer you are. 

If you have more on this topic I can help move it forward administratively. Please feel free to either ELI5 your request or provide code.

Thanks,

Aaron

**Comment by orlitzky** [comment]
I saw, thank you. I remember getting frustrated with this issue pretty quickly which probably explains my terse description. 

I think there were several issues intertwined, but the big one is that the ndoutils `./configure` script has been committed to the repository, and is being hacked on at each release rather than regenerated automatically (see for example https://github.com/NagiosEnterprises/ndoutils/commit/7406c6856e53fb45b64bf77075ed8293c85a5486). It should instead be possible to generate `./configure` automatically from `configure.ac` by running `autoreconf -fi`. And since `./configure` is automatically generated, there is usually no need to commit it to git. When a new release is made, the usual `autoreconf -fi; ./configure; make dist` process creates a `./configure` script and sticks it in the tarball. A human can do this, but github is capable of doing it with each tag as well.

The reason this suggestion is not straightforward, though, is because I already tried that and it doesn't work:

```
$ autoreconf -fi
ac-wrapper: autoreconf: warning: auto-detected versions not found ( 2.69 ); falling back to latest available
am-wrapper: aclocal: warning: auto-detected versions not found (1.9 ); falling back to latest available
configure.ac:25: warning: The macro 'AC_CONFIG_HEADER' is obsolete.
configure.ac:25: You should run autoupdate.
./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
configure.ac:25: the top level
configure.ac:52: warning: The macro 'AC_NAGIOS_GET_OS' is obsolete.
configure.ac:52: You should run autoupdate.
macros/ax_nagios_get_os:41: AC_NAGIOS_GET_OS is expanded from...
configure.ac:52: the top level
configure.ac:52: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:52: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
macros/ax_nagios_get_os:42: AX_NAGIOS_GET_OS is expanded from...
macros/ax_nagios_get_os:41: AC_NAGIOS_GET_OS is expanded from...
configure.ac:52: the top level
configure.ac:53: warning: The macro 'AC_NAGIOS_GET_DISTRIB_TYPE' is obsolete.
configure.ac:53: You should run autoupdate.
macros/ax_nagios_get_distrib:47: AC_NAGIOS_GET_DISTRIB_TYPE is expanded from...
configure.ac:53: the top level
configure.ac:53: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:53: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
macros/ax_nagios_get_distrib:48: AX_NAGIOS_GET_DISTRIB_TYPE is expanded from...
macros/ax_nagios_get_distrib:47: AC_NAGIOS_GET_DISTRIB_TYPE is expanded from...
configure.ac:53: the top level
configure.ac:54: warning: The macro 'AC_NAGIOS_GET_INIT' is obsolete.
configure.ac:54: You should run autoupdate.
macros/ax_nagios_get_init:53: AC_NAGIOS_GET_INIT is expanded from...
configure.ac:54: the top level
configure.ac:54: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:54: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
macros/ax_nagios_get_init:54: AX_NAGIOS_GET_INIT is expanded from...
macros/ax_nagios_get_init:53: AC_NAGIOS_GET_INIT is expanded from...
configure.ac:54: the top level
configure.ac:55: warning: The macro 'AC_NAGIOS_GET_INETD' is obsolete.
configure.ac:55: You should run autoupdate.
macros/ax_nagios_get_inetd:51: AC_NAGIOS_GET_INETD is expanded from...
configure.ac:55: the top level
configure.ac:55: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:55: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
macros/ax_nagios_get_inetd:52: AX_NAGIOS_GET_INETD is expanded from...
macros/ax_nagios_get_inetd:51: AC_NAGIOS_GET_INETD is expanded from...
configure.ac:55: the top level
configure.ac:56: warning: The macro 'AC_NAGIOS_GET_PATHS' is obsolete.
configure.ac:56: You should run autoupdate.
macros/ax_nagios_get_paths:46: AC_NAGIOS_GET_PATHS is expanded from...
configure.ac:56: the top level
configure.ac:56: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:56: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
macros/ax_nagios_get_paths:47: AX_NAGIOS_GET_PATHS is expanded from...
macros/ax_nagios_get_paths:46: AC_NAGIOS_GET_PATHS is expanded from...
configure.ac:56: the top level
configure.ac:57: warning: The macro 'AC_NAGIOS_GET_FILES' is obsolete.
configure.ac:57: You should run autoupdate.
macros/ax_nagios_get_files:44: AC_NAGIOS_GET_FILES is expanded from...
configure.ac:57: the top level
configure.ac:69: warning: The macro 'AC_HEADER_STDC' is obsolete.
configure.ac:69: You should run autoupdate.
./lib/autoconf/headers.m4:663: AC_HEADER_STDC is expanded from...
configure.ac:69: the top level
configure.ac:70: warning: The macro 'AC_HEADER_TIME' is obsolete.
configure.ac:70: You should run autoupdate.
./lib/autoconf/headers.m4:702: AC_HEADER_TIME is expanded from...
configure.ac:70: the top level
configure.ac:80: warning: The macro 'AC_TYPE_SIGNAL' is obsolete.
configure.ac:80: You should run autoupdate.
./lib/autoconf/types.m4:805: AC_TYPE_SIGNAL is expanded from...
configure.ac:80: the top level
configure.ac:142: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:142: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:142: the top level
configure.ac:150: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:150: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
configure.ac:150: the top level
configure.ac:165: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:165: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
configure.ac:165: the top level
configure.ac:186: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:186: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
configure.ac:186: the top level
configure.ac:258: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:258: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:258: the top level
configure.ac:316: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:316: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:316: the top level
configure.ac:317: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:317: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:317: the top level
configure.ac:324: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:324: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:324: the top level
configure.ac:332: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.ac:332: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
configure.ac:332: the top level
configure.ac:343: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:343: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:343: the top level
configure.ac:355: warning: The macro 'AC_NAGIOS_GET_SSL' is obsolete.
configure.ac:355: You should run autoupdate.
macros/ax_nagios_get_ssl:43: AC_NAGIOS_GET_SSL is expanded from...
configure.ac:355: the top level
configure.ac:363: warning: The macro 'AC_HELP_STRING' is obsolete.
configure.ac:363: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
configure.ac:363: the top level
autoheader-2.72: warning: missing template: HAVE_LIBWRAP
autoheader-2.72: warning: Use AC_DEFINE([HAVE_LIBWRAP], [], [Description])
autoheader-2.72: warning: missing template: HAVE_SUN_LEN
autoheader-2.72: warning: missing template: HAVE_SYSTEMD
autoheader-2.72: warning: missing template: INT32_T_IS_UINT
autoheader-2.72: warning: missing template: INT32_T_IS_ULONG
autoheader-2.72: warning: missing template: INT32_T_IS_USHORT
autoheader-2.72: warning: missing template: SOCKET_SIZE_TYPE
autoheader-2.72: warning: missing template: USE_MYSQL
autoheader-2.72: warning: missing template: USE_NANOSLEEP
autoheader-2.72: warning: missing template: USE_SSL_DH
autoheader-2.72: warning: missing template: U_INT32_T_IS_UINT
autoheader-2.72: warning: missing template: U_INT32_T_IS_UINT32_T
autoheader-2.72: warning: missing template: U_INT32_T_IS_ULONG
autoheader-2.72: warning: missing template: U_INT32_T_IS_USHORT
autoreconf-2.72: error: /usr/bin/autoheader-2.72 failed with exit status: 1
```

Most of the warnings can be ignored (or are easy to fix), but I don't know where the error is coming from. I'm sure I could figure it out eventually but it has the potential to become an ordeal.

**Comment by aaronagios** [comment]
Fair enough. If you find yourself inspired enough to endure the ordeal to sort it we can likely get it pushed through.

In the mean time I will get the current release process engaged, I'd hope to have a release of 2.15 next week.
